src/algebra/ordered_group.lean

```diff
@@ -301,6 +305,13 @@ instance [add_monoid α] : add_monoid (with_top α) :=
   add := (+),
   ..@additive.add_monoid _ $ @with_zero.monoid (multiplicative α) _ }
 
+@[simp, norm_cast] lemma coe_zero {α : Type*}
+  [add_monoid α] : ((0 : α) : with_top α) = 0 := rfl
+
+@[simp, norm_cast] lemma coe_eq_zero {α : Type*}
```

Comment:
If this lemma is proved by `norm_cast` it probably shouldn't need to be tagged `norm_cast` itself. Does something fail without this? what `norm_cast` lemmas does it use? Are other lemmas in the file provable with `norm_cast` already?

Reply:
Answers: 1) Nothing seems to fail if I untag it 2) The only way I know how to answer this is with `set_option trace.simplify.rewrite true`, which tells me coe_eq_zero is proved using coe_zero. There must be more to it than this though; norm_cast is doing something simp isn't doing. 3) with_bot.coe_zero can be proved by norm_cast